packages/block-editor/src/components/block-caption/index.native.js

```diff
@@ -44,16 +45,29 @@ const BlockCaption = ( {
 
 export default compose( [
 	withSelect( ( select, { clientId } ) => {
-		const { getBlockAttributes, getSelectedBlockClientId } = select(
-			blockEditorStore
-		);
+		const {
+			getBlockAttributes,
+			getSelectedBlockClientId,
+			getBlockName,
+			getBlockRootClientId,
+		} = select( blockEditorStore );
 		const { caption } = getBlockAttributes( clientId ) || {};
 		const isBlockSelected = getSelectedBlockClientId() === clientId;
 
+		const parentId = getBlockRootClientId( clientId );
+		const parentBlockName = getBlockName( parentId );
+
+		const hideCaption = hasBlockSupport(
```

Comment:
@SiobhyB, does this only affect `BlockCaption`s used in child blocks and not block captions found in regular blocks? For example, I tested the old Gallery block with this blocks support set to `true` and while the captions were still shown (which is correct behavior IMO), I wanted to verify this with you.

I see `getBlockRootClientId` [returns an empty string for top-level blocks](https://github.com/WordPress/gutenberg/blob/7ac99590ca14c5e5a800eae0a2c65a4a5b26c4f0/docs/reference-guides/data/data-core-block-editor.md#getblockrootclientid), so the check here should default to `false` which will show the caption, which makes sense.

Reply:
@guarani, that's right, regular blocks shouldn't be impacted by this change. I took the opportunity to add a comment to the code in https://github.com/WordPress/gutenberg/pull/36618/commits/06bfad93f5f9f173fc42dfa8f7af57aeef98ec99 as I think it could use clarification LMK if you think it could use any tweaks. :)

Reply:
Cool! 👍